### DISCUSSION TITLE
[Quest] Allow for sync when app is in the Background or  On resume

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#1732] [Ideas]
**Describe the bug**
App enters a permanent sync state without completing the initial sync, once the user clicks on home or the app resumes from the background state. Seem like the app tries to resync the data afresh on every resume instead of picking up from where it left 

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to the Quest eCBIS app
2. Click on  the launcher icon (Ist time launch)
3. Wait for the sync to start and then launch a new app or go back to the home screen 
4. Launch the app once again 
5. See the error 

**Expected behaviour**
The app should be able to sync data once initiated and resume on app-resume in case the app was switched to another app or pushed to the background on press home 

**Screenshots**
N/A

**Smartphone (please complete the following information):**
 - Device: Pixel 4a
 - Android version: 13
 - Build number: TP1A.220905.004

**Additional context**
The user should get prompts on 
1. Sync initiated
2. Sync resumed in case the app does not have the last sync timestamp on 1st launch  or on resume from the background state
3. Sync succeeded/failed

**Comment by DebbieArita** [comment]
I observed that this is true, the app resyncs the data afresh on every resume. Investigating further on how to resolve this.

**Comment by pld** [comment]
Can the app sync in the background?

**Comment by f-odhiambo** [comment]
> Can the app sync in the background?

At the moment this is not the case. Based on the architecture in place, once the app goes to the background the activity is closed/stopped

**Comment by pld** [comment]
Can we look into this, what would it take to add background sync?

**Comment by DebbieArita** [comment]
From the discussions and findings so far... I see that...
1.For the initial sync (that is sync after login), app activity has to be on the foreground/one has to remain on the app for sync to complete.
2. Sync process does keep running in the background, as shown by the dialog box continuously running, and the sync states are logged on the application, but we don't have a way to save the different sync states as sync is ongoing. As a result, I see that when you leave the app while running the initial sync, on every resumption, incoming data appears to be syncing again until it completes...
3. Sync states are remitted on a MutableSharedFlow defined in the FHIR SDK...
4. We need further discussions on how to improve upon this...
